04-store-api/starter/controllers/products.js

```diff
@@ -0,0 +1,67 @@
+const Product = require('../models/product');
+
+const getAllProductsStatic = async (req, res) => {
+  const products = await Product.find({ price: { $gt: 30 } })
+    .select('name price')
+    .limit(10)
+    .skip(5);
+  res.status(200).json({ products, nbHits: products.length });
+};
+
+const getAllProducts = async (req, res) => {
+  const { featured, company, name, sort, fields, numericFilters } = req.query;
+  const queryObject = {};
+  if (featured) {
+    queryObject.featured = featured === 'true' ? true : false;
+  }
+  if (company) {
+    queryObject.company = company;
+  }
+  if (name) {
+    queryObject.name = { $regex: name, $options: 'i' };
```

Comment:
Could anything go wrong if we allow the client to send us a regex?

Reply:
Absolutely, allowing the client to send a regex can be a security risk as it can potentially lead to a vulnerability known as Regular Expression Denial of Service (ReDoS) which can occur when processing user input that contains a maliciously crafted regular expression. To mitigate these attacks, it's important to properly validate and sanitize any user input using 3rd party libraries or built-in methods that limit the complexity of regular expression.